src/install/install.zig

```diff
@@ -4459,7 +4459,17 @@ pub const PackageManager = struct {
                 return try std.fs.cwd().makeOpenPathIterable(path, .{});
             }
 
-            if (bun.getenvZ("XDG_CACHE_HOME") orelse bun.getenvZ("HOME")) |home_dir| {
+            if (bun.getenvZ("XDG_CACHE_HOME")) |home_dir| {
+                var buf: [bun.MAX_PATH_BYTES]u8 = undefined;
+                var parts = [_]string{
+                    "bun",
+                    "bin",
+                };
+                var path = Path.joinAbsStringBuf(home_dir, &buf, &parts, .auto);
+                return try std.fs.cwd().makeOpenPathIterable(path, .{});
+            }
+
+            if (orelse bun.getenvZ("HOME")) |home_dir| {
                 var buf: [bun.MAX_PATH_BYTES]u8 = undefined;
                 var parts = [_]string{
                     ".bun",
```

Comment:
I can't suggest the exact change since it goes outside the diff context, but this should be:

```zig
var parts = [_]string{ ".local", "bin", "bun" };
```

Reply:
This would move the cache outside of the default bun dir for users who don't have `$XDG_CACHE_HOME` set.

I'd be in favor of that, but is that desirable to the `bun` team?